x-pack/plugins/security_solution/server/lists_integration/endpoint/validators/blocklist_validator.ts

```diff
@@ -0,0 +1,275 @@
+          isValidHash(hash) ? undefined : `invalid hash value [${hash}]`,
+      })
+    ),
+    schema.conditional(
+      schema.siblingRef('field'),
+      FileExecutablePath,
+      schema.arrayOf(
+        schema.string({
+          validate: (pathValue: string) =>
+            pathValue.length > 0 ? undefined : `invalid path value [${pathValue}]`,
+        })
+      ),
+      schema.arrayOf(
+        schema.string({
+          validate: (signerValue: string) =>
+            signerValue.length > 0 ? undefined : `invalid signer value [${signerValue}]`,
+        })
+      )
+    )
+  ),
+};
+
+// Windows Signer entries use a Nested field that checks to ensure
+// that the certificate is trusted
+const WindowsSignerEntrySchema = schema.object({
+  type: schema.literal('nested'),
+  field: FileCodeSigner,
+  entries: schema.arrayOf(
+    schema.object({
+      field: schema.literal('subject_name'),
+      value: schema.arrayOf(schema.string({ minLength: 1 })),
+      type: schema.literal('match_any'),
+      operator: schema.literal('included'),
+    })
+  ),
+});
+
+const WindowsEntrySchema = schema.oneOf([
+  WindowsSignerEntrySchema,
+  schema.object({
+    ...CommonEntrySchema,
+    field: schema.oneOf([FileHashField, FileExecutablePath]),
+  }),
+]);
+
+const LinuxEntrySchema = schema.object({
+  ...CommonEntrySchema,
+});
+
+const MacEntrySchema = schema.object({
+  ...CommonEntrySchema,
+});
+
+// Hash entries validator method.
+const hashEntriesValidation = (entries: BlocklistConditionEntry[], allowedHashes: string[]) => {
+  const currentHashes = entries.map((entry) => entry.field);
+  // If there are more hashes than allowed (three) then return an error
+  if (currentHashes.length > allowedHashes.length) {
+    const allowedHashesMessage = allowedHashes
+      .map((hash) => hash.replace('file.hash.', ''))
+      .join(',');
+    return `There are more hash types than allowed [${allowedHashesMessage}]`;
+  }
+
+  const hashesCount: { [key: string]: boolean } = {};
+  const duplicatedHashes: string[] = [];
+  const invalidHash: string[] = [];
+
+  // Check hash entries individually
+  currentHashes.forEach((hash) => {
+    if (!allowedHashes.includes(hash)) invalidHash.push(hash);
+    if (hashesCount[hash]) {
+      duplicatedHashes.push(hash);
+    } else {
+      hashesCount[hash] = true;
+    }
+  });
+
+  // There is more than one entry with the same hash type
+  if (duplicatedHashes.length) {
+    return `There are some duplicated hashes: ${duplicatedHashes.join(',')}`;
+  }
+
+  // There is an entry with an invalid hash type
+  if (invalidHash.length) {
+    return `There are some invalid fields for hash type: ${invalidHash.join(',')}`;
+  }
+};
+
+// Validate there is only one entry when signer or path and the allowed entries for hashes
+const entriesSchemaOptions = {
+  minSize: 1,
+  validate(entries: BlocklistConditionEntry[]) {
+    const allowedHashes = ['file.hash.md5', 'file.hash.sha1', 'file.hash.sha256'];
```

Comment:
Could promote this up to a module level readonly `const` and maybe even avoid some of the duplication with the values used in `FileHashField` above.